src/dropRepeatsWith.js

```diff
@@ -0,0 +1,42 @@
+var _curry2 = require('./internal/_curry2');
+var _dispatchable = require('./internal/_dispatchable');
+var _xdropRepeatsWith = require('./internal/_xdropRepeatsWith');
+var last = require('./last');
+
+
+/**
+ * Returns a new list without any consecutively repeating elements. Equality is
+ * determined by applying the supplied predicate two consecutive elements.
+ * The first element in a series of equal element is the one being preserved.
+ *
+ * Acts as a transducer if a transformer is given in list position.
+ * @see R.transduce
+ *
+ * @func
+ * @memberOf R
+ * @category List
+ * @sig [a] -> [a]
+ * @param {Function} pred A predicate used to test whether two items are equal.
+ * @param {Array} list The array to consider.
+ * @return {Array} `list` without repeating elements.
+ * @example
+ *
+ *      function lengthEq(x, y) { return Math.abs(x) === Math.abs(y); };
+ *      var l = [1, -1, 1, 3, 4, -4, -4, -5, 5, 3, 3];
+ *      R.dropRepeatsWith(lengthEq, l); //=> [1, 3, 4, -5, 3]
+ */
+module.exports = _curry2(_dispatchable('dropRepeatsWith', _xdropRepeatsWith, function dropRepeatsWith(pred, list) {
+  var result = [];
+  var idx = 0;
+  if (list.length === 0) {
+    return result;
+  }
+  result.push(list[0]);
+  while (++idx < list.length) {
+    if (!pred(last(result), list[idx])) {
```

Comment:
Would it be cleaner to cache the previous call rather than call `last` on each iteration?